rust/src/ortc.rs

```diff
@@ -641,24 +641,25 @@ pub(crate) fn can_consume(
 /// codecs' RTCP feedback and header extensions, and also enables or disabled RTX.
 #[allow(clippy::suspicious_operation_groupings)]
 pub(crate) fn get_consumer_rtp_parameters(
-    consumable_params: &RtpParameters,
-    caps: &RtpCapabilities,
+    consumable_rtp_parameters: &RtpParameters,
+    remote_rtp_capabilities: &RtpCapabilities,
     pipe: bool,
+    enable_rtx: bool,
 ) -> Result<RtpParameters, ConsumerRtpParametersError> {
     let mut consumer_params = RtpParameters {
-        rtcp: consumable_params.rtcp.clone(),
+        rtcp: consumable_rtp_parameters.rtcp.clone(),
         ..RtpParameters::default()
     };
 
-    for cap_codec in &caps.codecs {
+    for cap_codec in &remote_rtp_capabilities.codecs {
         validate_rtp_codec_capability(cap_codec)
             .map_err(ConsumerRtpParametersError::InvalidCapabilities)?;
     }
 
     let mut rtx_supported = false;
 
-    for mut codec in consumable_params.codecs.clone() {
-        if let Some(matched_cap_codec) = caps
+    for mut codec in consumable_rtp_parameters.codecs.clone() {
```

Comment:
@nazar-pc I need that this changes to be in sync with Node changes in ortc.ts in `getConsumerRtpParameters()`:

```ts
for (const codec of consumableCodecs)
	{
		const matchedCapCodec = remoteRtpCapabilities.codecs!
			.find((capCodec) => matchCodecs(capCodec, codec, { strict: true }));

		if (!matchedCapCodec)
		{
			continue;
		}

		if (!enableRtx && isRtxCodec(codec))
		{
			continue;
		}

		codec.rtcpFeedback = matchedCapCodec.rtcpFeedback!
			.filter((fb) => (
				(enableRtx || fb.type !== 'nack' || fb.parameter)
			));

		consumerParams.codecs.push(codec);
	}
```

Any help please? I'm a bit lost here.

Reply:
`if` can be added to the beginning of the `for` loop the same way as in TypeScript (there is `.is_rtx()` call used in this file).

Then in `*codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();` you can replace that with something like this (may not compile):
```rust
*codec.rtcp_feedback_mut() = matched_cap_codec
    .rtcp_feedback()
    .iter()
    .filter(|fb| do checks here)
    .cloned()
    .collect();
```

Examples of matching of feedback type also present in this file.

Reply:
Done in https://github.com/versatica/mediasoup/pull/1015/commits/09b6c085e7919cd2539cb0715dc639974b26f5ce